### DISCUSSION TITLE
Will yfiles-jupyter-graphs work for kuzu?

### DISCUSSION THREAD

**Discussion by Analect** [yWorks/yfiles-jupyter-graphs#41] [Q&A] [answered]
[kuzu](https://kuzudb.com/docusaurus/getting-started/) is an embedded graph database that can run within Colab. There is a demo notebook [here](https://colab.research.google.com/drive/15OLPggnRSBmR_K9yzq6iAGE5MDzNwqoN). I was wondering if the yfiles-jupyter-graphs will work with it, given that it supports cypher. Thanks.

**Comment by yGuy** [comment]
Yes, it will. 
The plugin does not depend or offer a deep integration with any *specific* datasources. There are some convenience functionalities available to make it easy to import in-memory graph data structures for various popular sources. But at its core, it works with simple python arrays and dictionaries. The "neo4j-integration" that's built-in is restricted to using the graph implementation that is returned by the query results. 
If you can use the neo4j python driver to connect to kuzu, you can simply use the "neo4j" examples. Otherwise, you can use any of the generic examples or convert your graph to one of the many popular supported graph data structures like networkx, etc.

**Reply by yGuy** [reply, marked as answer]
This turns out to be pretty easy via the "get_as_networkx()" functionality of Kuzu:

See my modified version of the notebook: https://colab.research.google.com/drive/1TveYBPKHPJ4fnVweHf6tNs44D82h0M82

![image](https://github.com/yWorks/yfiles-jupyter-graphs/assets/2486305/edd62d09-5840-40bf-b469-841c79165eb6)

```
!pip install kuzu==0.0.7 yfiles-jupyter-graphs
```

```
from yfiles_jupyter_graphs import GraphWidget
from google.colab import output
output.enable_custom_widget_manager()
result = conn.execute('MATCH (a:User) - [f:Follows] -> (b:User) RETURN a,f,b')
GraphWidget(graph = result.get_as_networkx())
```

**Reply by Analect** [reply]
@yGuy ... thanks for your helpful and thorough response. Great tooling.
